Shopify/cli
# Add --strict flag to theme push to enforce theme check before pushing

### WHY are these changes introduced?

Adds a new `--strict` flag to the theme push command that enforces theme check validation before allowing files to be pushed to a remote theme.

### WHAT is this pull request doing?

Adds a new `--strict` flag to the `theme push` command that:
- Runs theme check validation before pushing files
- Blocks the push operation if any theme check errors are found (warnings are allowed)
- Displays offense information and summary to help developers fix issues
- Passes the output format down to theme check if JSON is specified
- Adds tests

### How to test your changes?

1. Create a theme with some theme check violations
2. Run `theme push --strict`
3. Verify the command fails and displays theme check offenses
4. Fix the violations
5. Run `theme push --strict` again
6. Verify the push succeeds with no violations

### Errors
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/JdHLnhebSbtZTZO01I1e/ee69b737-f31f-4613-87ce-896a8bce119c.png)

### Warnings
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/JdHLnhebSbtZTZO01I1e/c6debd98-bd2d-4467-88ea-ef3ebfa0306d.png)

### Measuring impact


### Checklist
